### PR TITLE
Simple Game Server Example: Upgrade Docker to 24.0.6

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Work on #3530

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by Kalaiselvi84** [comment]
Here's the failed cloud-build log - https://gist.github.com/Kalaiselvi84/2abc327822338b70e3e082da45ff42fa

**Comment by Kalaiselvi84** [comment]
Build has passed with `--use` flag in this line: `-DOCKER_CLI_EXPERIMENTAL=enabled docker buildx create --name=$(BUILDX_WINDOWS_BUILDER) --use`

Log - https://gist.github.com/Kalaiselvi84/6b02ec26b5c8d4c0362060f81e31cd29

If this looks good, let me know the next steps please.

File: examples/simple-game-server/Makefile
Comment (markmandel) [inline examples/simple-game-server/Makefile:90]: Just noting we'll want to increment the version number as well in here, since this will be a new build (then we will want to update the simple-game-server version everywhere as well - likely in a separate PR).

**Review by markmandel — COMMENTED** [review]
Approach looks good - is this successfully building the image and container?

**Comment by Kalaiselvi84** [comment]
> Approach looks good - is this successfully building the image and container?

Build has passed with `https://github.com/googleforgames/agones/blob/main/examples/simple-game-server/Makefile#L122`

Failed Scenarios:
The build failed without using the `--use` flag and `DOCKER_CLI_EXPERIMENTAL=enabled`. Log: https://gist.github.com/Kalaiselvi84/38a273a6dbb7e308fdb4be166fc597fd

Without `--use` flag: https://gist.github.com/Kalaiselvi84/2abc327822338b70e3e082da45ff42fa

Without `DOCKER_CLI_EXPERIMENTAL=enabled`: https://gist.github.com/Kalaiselvi84/0d0b5fcc8de6fb9f31ac0d8588f40fda

**Comment by markmandel** [comment]
My apologies, I'm not following - does it work as it is currently written in the PR?

**Comment by Kalaiselvi84** [comment]
> My apologies, I'm not following - does it work as it is currently written in the PR?

Sorry for the confusion, it isn't working

**Comment by markmandel** [comment]
Just playing with this myself - there is definitely some weird stuff going on... 

For future reference:

Locally I'm running Docker 24.x (client and server), but on Cloud Build, we have a 24.x Client on a 20.x Docker server.

I think that is why we get weirdness on Cloud Build, when it seems to be fine building locally.

> The build failed without using the --use flag and DOCKER_CLI_EXPERIMENTAL=enabled in https://github.com/googleforgames/agones/blob/main/examples/simple-game-server/Makefile#L122

I think I've just ended up in exactly what you were originally trying to say 🤦🏻 just doing some tests to ensure it works across both versions.

**Comment by Kalaiselvi84** [comment]
> Locally I'm running Docker 24.x (client and server), but on Cloud Build, we have a 24.x Client on a 20.x Docker server.

Got it..
Does the Google Cloud team have any plans to upgrade the Docker server to version 24.x in the Cloud Build env?
Next step please?

File: examples/simple-game-server/Makefile
Comment (markmandel) [inline examples/simple-game-server/Makefile:108]:
```suggestion
	cd $(root_path) && DOCKER_CLI_EXPERIMENTAL=enabled \
```

File: examples/simple-game-server/Makefile
Comment (markmandel) [inline examples/simple-game-server/Makefile:115]:
```suggestion
	-docker buildx create --name=$(BUILDX_WINDOWS_BUILDER) --use
```

**Review by markmandel — COMMENTED** [review]
This is what I ended up with that worked across local Docker, 24.x and Cloud Build -- does this work for you too? If so, let's run with it, and update the issue for these instructions.

**Comment by markmandel** [comment]
Just also making a note that after this change (and pushing the new image to prod), will need to change all the other references to `simple-game-server:0.22` - such as the website, the e2e framework, etc.

File: examples/simple-game-server/Makefile
Comment (Kalaiselvi84) [reply examples/simple-game-server/Makefile:115]: I am going to follow what you're suggesting. Just a thought – should we aim for uniformity between the `simple-game-server` and `Xonotic`?
https://github.com/googleforgames/agones/blob/main/examples/xonotic/Makefile#L86-L93

Comment (Kalaiselvi84) [reply examples/simple-game-server/Makefile]: Success log: https://gist.github.com/Kalaiselvi84/e6dddbfb6582b563ebb9d5dc2efb43e2

**Comment by Kalaiselvi84** [comment]
> Just also making a note that after this change (and pushing the new image to prod), will need to change all the other references to `simple-game-server:0.22` - such as the website, the e2e framework, etc.

Shall I push the latest tag to prod now or should I do it after this PR is merged?

**Comment by markmandel** [comment]
> > Just also making a note that after this change (and pushing the new image to prod), will need to change all the other references to `simple-game-server:0.22` - such as the website, the e2e framework, etc.
> 
> Shall I push the latest tag to prod now or should I do it after this PR is merged?

Either or. I'll approve this now, so push it when it's ready, and then can do the update to the rest in a separate PR.

**Review by markmandel — APPROVED** [review]

**Pull request merged**
